### PR TITLE
Improve cargo-raze usage

### PR DESCRIPTION
**Pull request by daviddrysdale**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [ ] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

**Comment by tiziano88** [comment]
We already have a top-level Cargo.lock file, we really should just be using that, rather than trying to replicate it. If there is no "native" way in cargo raze, perhaps we can trick it by using symlinks or something like that?

**Comment by daviddrysdale** [comment]
Shifted to make `cargo/Cargo.toml` part of the top level workspace.  Normal `cargo` operations now give a warning `unused manifest key: raze` that I can't see an easy way to get rid of.

File: scripts/cargo_raze [hunk 1]
```diff
@@ -5,4 +5,5 @@ readonly SCRIPTS_DIR="$(dirname "$0")"
 source "$SCRIPTS_DIR/common"
 
 cd cargo
+rm -f remote/*
```
Comment (tiziano88) [inline scripts/cargo_raze:8]: Could you explain why this is needed?

Comment (daviddrysdale) [reply scripts/cargo_raze:10]: Added a comment.

**Comment by tiziano88** [comment]
Ideally what we need is a way of pointing `cargo-raze` to an existing `Cargo.lock` file, and generate everything from that. If that's not something that `cargo-raze` does already, perhaps we could contribute to it? It seems all the building blocks should be around already. @daviddrysdale WDYT?

**Comment by daviddrysdale** [comment]
I wonder if I'm just holding `cargo-raze` wrongly somehow.  But yeah, if not then asking it to choose dep versions based on an existing `Cargo.lock` seems helpful.

**Comment by daviddrysdale** [comment]
Moved this PR to just have the `rm cargo/remote/*` and raised #875.

**Pull request merged**